ast/modify.go

```diff
@@ -1,55 +1,121 @@
 package ast
 
-import "fortio.org/log"
+import "fmt"
 
 // Note, this is somewhat similar to eval.go's eval... both are "apply"ing.
-func Modify(node Node, f func(Node) Node) Node {
+func Modify(node Node, f func(Node) Node) Node { //nolint:funlen,gocyclo // yeah lots of types.
 	// TODO: add err checks for _s.
```

Comment:
What about a map[string]func(node)

Where you use fmt.Sprintf("%T", node) as a key and ask if the node type is in the map, if not, panic, if present call the func?

This way you will have one method per type.

Maybe it's naive, and may need a type assertion in each method, but you could split this gigantic piece of code

Reply:
I wouldn't use a string version and keep attaching strings no - now maybe a map can have a type as key, not sure

the code is very mechanical (copy for each type and recurse on each components), a few lines in each case so I don't think the linters alone should be a reason to break it out